P/PAPI/build_tarballs.jl

```diff
@@ -1,28 +1,70 @@
 # Note that this script can accept some limited command-line arguments, run
 # `julia build_tarballs.jl --help` to see a usage message.
-using BinaryBuilder, Pkg
+using BinaryBuilder, Pkg, BinaryBuilderBase
+
+const YGGDRASIL_DIR = "../.."
+include(joinpath(YGGDRASIL_DIR, "fancy_toys.jl"))
+include(joinpath(YGGDRASIL_DIR, "platforms", "cuda.jl"))
 
 name = "PAPI"
-version = v"6.0.0"
+version = v"7.0.0"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://bitbucket.org/icl/papi.git", "05a4d383eea251db990ab668ca4a13a74427a152")
+    GitSource("https://bitbucket.org/icl/papi.git", "de96060998cd9fc77396c5e100e52e0ea1cdc3c3"),
+    DirectorySource("./bundled")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir
-cd papi/src
+cd $WORKSPACE/srcdir/papi
+
+# Apply all our patches
+if [ -d $WORKSPACE/srcdir/patches ]; then
+for f in $WORKSPACE/srcdir/patches/*.patch; do
+    echo "Applying patch ${f}"
+    atomic_patch -p1 ${f}
+done
+fi
+
+cd src
 if [[ "${target}" == *-musl* ]]; then
     CFLAGS="-D_GNU_SOURCE"
 fi
+
+export PAPI_CUDA_ROOT="${prefix}/cuda"
+COMPONENTS=()
+if [[ -d "${prefix}/cuda" ]]; then
+    COMPONENTS+=(cuda)
+fi
+
+if [[ "${target} == "powerpc64le-*"" ]]; then
+  CPU=POWER8
+elif [[ "${target}" == "x86_64-*" || "${target}" == "i686-*" ]]; then
```

Comment:
You can also use
```suggestion
elif [[ "${proc_family}" == "intel" ]]; then
```
see https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables.  Also, I'm not sure the `*` wildcard plays well with quoting there.